applications/FluidDynamicsHydraulicsApplication/custom_utilities/hydraulic_fluid_auxiliary_utilities.h

```diff
@@ -0,0 +1,151 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Uxue Chasco
+
+#if !defined(KRATOS_HYDRAULIC_FLUID_AUXILIARY_UTILITIES_H)
+#define KRATOS_HYDRAULIC_FLUID_AUXILIARY_UTILITIES_H
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometries/geometry.h"
+#include "includes/node.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "modified_shape_functions/modified_shape_functions.h"
+#include "includes/global_pointer_variables.h"
+
+// Application includes
+#include "hydraulic_fluid_auxiliary_utilities.h"
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsHydraulicsApplication
+///@{
+
+///@name Kratos classes
+///@{
+
+class KRATOS_API(FLUID_DYNAMICS_HYDRAULICS_APPLICATION) HydraulicFluidAuxiliaryUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using SizeType = std::size_t;
+
+    using IndexType = std::size_t;
+
+    using NodeType = Node;
+
+    using GeometryType = Geometry<NodeType>;
+
+    using PointsArrayType = typename GeometryType::PointsArrayType;
+
+    using ModifiedShapeFunctionsFactoryType = std::function<ModifiedShapeFunctions::UniquePointer(const GeometryType::Pointer, const Vector&)>;
+
+    ///@}
+    ///@name Static Operations
+    ///@{
+    /**
+     * @brief This functions calculates the wetted perimeter of a given condition in order to apply the
+     * corresponding inlet boundary condition.
+     * @param rModelPart Inlet Model Part
+     * @param rSkinFlag Flag that marks the conditions to be included in the calculation
+     * @param rDistanceVariable Reference to the variable containing the distance
+     * @param IsHistorical True if the distance is in the historical database, false otherwise
+     * @return
```

Comment:
```suggestion
     * @return The wetted perimeter value
```
I think you also missed the ones in the other methods.